### ISSUE TITLE
Conditional rendering causes the Dialog to close immediately in controlled component

### ISSUE THREAD

**Issue by erm1116** [chakra-ui/ark#2744]
### Description

When rendering Dialog conditionally, Dialog component wil be closed Immediately.
(Sometimes, it appears correctly.)

I tried conditional rendering like below
```tsx
export default function App() {
  const [showModal, setShowModal] = useState(false);
  return (
    <div className="App">
      <button onClick={() => setShowModal(true)}>Open Modal</button>
      {showModal && (
	    <Dialog.Root open={showModal} onOpenChange={(e) => setShowModal(e.open)}>
	      <Portal>
	        <Dialog.Backdrop />
	        <Dialog.Positioner>
	          <Dialog.Content>
	            <Dialog.Title>Test</Dialog.Title>
	            <Dialog.Description>Hello</Dialog.Description>
	            <Dialog.CloseTrigger onClick={setShowModal(false)}>Close</Dialog.CloseTrigger>
	          </Dialog.Content>
	        </Dialog.Positioner>
	      </Portal>
	    </Dialog.Root>
      )}
    </div>
  );
}
```

### Link to Reproduction (or Detailed Explanation)

https://codesandbox.io/p/sandbox/arkui-dialog-forked-ckdfl7

### Steps to Reproduce

1. click `Open Dialog` button
2. Dialog appears just for a moment and will be disappeared

Screen capture: 

https://github.com/user-attachments/assets/9857c589-a112-44fe-8db5-43ad95d24a6f

### Ark UI Version

3.6.2 (latest)

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Browser

Arc Browser 1.54.0 (Chromium Engine 127)

**Comment by segunadebayo** [comment]
I don't recommend conditionally rendering the Dialog.

Please use `unmountOnExit` and `lazyMount` instead.

**Comment by erm1116** [comment]
@segunadebayo 
I use ArkUI as the basis for a component library.
As a component library, it is of course possible to prevent conditional rendering when creating a `Dialog` component, but I think it is difficult to impose such restrictions on users of component library.

Should this be prevented by providing custom lint rules?
In my opinion, I thought it would be better to make it possible to display without problems when conditional rendering is used, because we don't know how it will be used.

**Comment by ivancuric** [comment]
I am getting another issue, where SolidJS' `Portal`  component as a child of `Dialog.Root` gets rendered even if the `open` prop is false, and `lazyMount` + `unmountOnExit` are enabled, leaving a zombie `<div>` element without any content.

The way I solved it was to wrap the entire `Dialog.Root` in a `<Show>` conditional render. Alternatively, I could just render the `Portal` conditionally, but it's unclear which mechanism is robust.

Especially considering how many different options there exist on the `Dialog.Root` that might impact the rendering logic, which aren't well explained in the documentation, such as:

> `present` — Whether the node is present (controlled by the user)

> `immediate` — Whether to synchronize the present change immediately or defer it to the next frame

I've tried fiddling with those as well without any success.

@erm1116 I generally found that ArkUI has a number of race conditions or multiple event dispatches which can be mitigated by using `setTimeouts` or [even combining with with clicking on stuff a few times](https://github.com/chakra-ui/ark/issues/2716#issuecomment-2374009239).

For example, pressing CMD+K on the documentation site opens two modals (check).
